### ISSUE TITLE
[BUG] Unable to sync repositories

### ISSUE THREAD

**Issue by Merennor** [Droid-ify/client#719]
**Describe the bug**
Repositories not synced with working WiFi (other apps working fine). Reinstalling doesn't help.
(**Everything worked fine until March 2024**)

**To Reproduce**
1. Open app
2. Click on 'Sync repositories'
3. See error in Notifications

**Expected behavior**
Apps appear in the list of available apps (Explore)

**Screenshots**
![image](https://github.com/Droid-ify/client/assets/105452638/9d4c64b7-2f8c-4045-9f4d-f91e50410ff2)

**Tablet:**
 - Device: Huawei MediaPad M3 Lite
 - OS: Android 7.0
 - Version: v0.6.3

**Additional context**
Logs while syncing:
[logcat.txt](https://github.com/Droid-ify/client/files/15048543/logcat.txt)
The same thing in Neo Store BUT not in official F-Droid client.

**Comment by msinfo32github** [comment]
Having the same problem here running Android 14 (GrapheneOS) on a Pixel 7. Apps that have updates available seem to appear in the updates tab even though an error is presented however.

**Comment by Iamlooker** [comment]
Can you try switching to a mirror of FDroid?

I am unable to reproduce this. And was this also the case in previous releases?

**Comment by Iamlooker** [comment]
Can any one try and get logs? I am unable to reproduce this issue, and cannot help without logs

**Comment by msinfo32github** [comment]
This seems to have been fixed on my device by updating Droid-ify and then next time I checked for updates (multiple days later) it seems to have been fixed.

**Comment by lazy-pupp** [comment]
My Droid-ify got uninstalled it appears. I'll check, get back after. :-)

Ivan Pavlović, prof.

sri, 24. tra 2024. 18:11 msinfo32 ***@***.***> je napisao:

**Comment by okgamr** [comment]
I got this problem too 
![Screenshot_2024-05-18-21-19-38-726-edit_com.looker.droidify.jpg](https://github.com/Droid-ify/client/assets/128509556/8a76762b-efd3-42ba-bfc6-6ba1c79f2fbd)

But F-Droid repo syncs, aka no errors there

**Comment by tsssu** [comment]
I think this issue related to isp, some provider blocks izzy and fdroid repo and adding DoH support would fix this issue. The reason why is because i also facing the same problem here.

**Comment by msinfo32github** [comment]
@tsssu Seems like you're correct in regards to an ISP issue - however I'm unsure if adding DoH will solve this? Are you using the 1.1.1.1 app? - I believe that's a VPN?

I've noticed recently this problem has been happening again for myself, and have noticed that navigating to some of the repositories itself has shown an error `PR_END_OF_FILE_ERROR`. I've been having some problems with my current ISP related to this error as well. However, on a second/third attempt on reloading the page it seems successful for every attempt after within a specified time period. (

I believe this is related to how my ISP uses CG-NAT (and not very well). It seems that maybe potentially adding a re-try by default (does this already exist?) or a configurable re-try in the settings could solve this problem for users with intermittent internet issues. - it seems like some users behind CG-NAT may have an IP address which changes often (e.g. every 30 seconds in worst cases) which causes this problem?

**Comment by tsssu** [comment]
> @tsssu Seems like you're correct in regards to an ISP issue - however I'm unsure if adding DoH will solve this? Are you using the 1.1.1.1 app? - I believe that's a VPN?

Yeah, 1.1.1.1 app is using VPN service. I tried to use Private DNS feature on android and then switching between Adguard and cloudflare, but the problem no longer exists even without enabling Private DNS. So i couldn't sure it would resolve this issues sorry, but i saw some app implemented DoH just in case if something wrong in some nations.

> It seems that maybe potentially adding a re-try by default (does this already exist?) or a configurable re-try in the settings could solve this problem for users with intermittent internet issues.

I would agree to that if you want to plan adding re-try button in the notification because so far i dont think there is any.

**Comment by pdimu** [comment]
Had this issue been fixed?

**Comment by UncleBob21** [comment]
Can confirm: NOT SOLVED
- On the same Wifi: 
mobile Lineage 7.1.2  (Droid-ify v0.6.3 ) works
tablet Android 7.0 (Droid-ify  v0.6.1, v0.6.2, v0.6.3) does NOT work
- its a permanent error and it applies to mostly all repositories!
- Tablet was working fine on v0.6.1 until some time ago!
- VPN connection has same error, tried several countries

- Repositories NOT working:
F-Droid
IzzyDroid
Guardian Project
Molly
Collobora
KDE Android
Divest OS
Umbrella
Nanodroid
Divest OS Unofficial
KDE Stables

- Repositories  working:
Netsyms
Calyx
patched-apps
Wind Offline
ThreemaLibre

- Suggestion: 
Possible source may be regarding an update and changes on the repositories side?

Please look into this

**Comment by manieKMP** [comment]
For Androids below 8.0 (or v. 6 and 7.0) the issue are too old/expired SSL certificates. It can be fixed if user will install a "new" ones from https://letsencrypt.org/certificates/ for example. Download pem files for root X1 and X2 from "Certificate details (self-signed)" row. Then go to system settings/options (gear icon) and somewhere under wifi, advanced settings find something called "install certificates" and choose downloaded pem files (if you can't find installation option try to use search bar in os settings and type "cert"...). Every installation of pem need to be "approved" by a pin code and named. After that droid-fy should be able to sync with repos without any problem.

**Comment by b-risk** [comment]
> For Androids below 8.0 (or v. 6 and 7.0) the issue are too old/expired SSL certificates. It can be fixed if user will install a "new" ones from https://letsencrypt.org/certificates/ for example. Download pem files for root X1 and X2 from "Certificate details (self-signed)" row. Then go to system settings/options (gear icon) and somewhere under wifi, advanced settings find something called "install certificates" and choose downloaded pem files (if you can't find installation option try to use search bar in os settings and type "cert"...). Every installation of pem need to be "approved" by a pin code and named. After that droid-fy should be able to sync with repos without any problem.

having the same problem and im on v0.6.4, installing these certificates and orbot didn't fix it for me, the phone im havng this issue with is on android 7 though

**Comment by Iamlooker** [comment]
I suggest trying: https://github.com/Droid-ify/client/issues/719#issuecomment-2439013092

If it does not work, open a new issue and provide logs

**Comment by DantesNoWar** [comment]
Hi, I have the same issue. I didn't get the solution from this discussion. Please, can someone help me fix this?

HyperOS: 1.0.8.0 - Android 14
Droid-Ify: v0.6.6

**Comment by nick-77** [comment]
> Hi, I have the sale issue. I didn't get the solution from this discussion. Please, can someone help me fix this?
> 
> HyperOS: 1.0.8.0 - Android 14
> Droid-Ify: v0.6.6
> 
Had the same issue, resetting wifi and Bluetooth settings fixed the issue

**Comment by DantesNoWar** [comment]
@nick-77 do you mean my device network settings?
I can't find anything like that on my phone.
Anyway, the problem is still there. 

![Image](https://github.com/user-attachments/assets/849d19d2-cad7-41f4-b379-83f31389e244)

**Comment by nick-77** [comment]
> @nick-77 do you mean my device network settings?
> I can't find anything like that on my phone.
> Anyway, the problem is still there. 
> 

The setting is usually under 'reset' options 
Attaching an image (samsung device)
![BkZzIEfNvnAISNtZT9b5r6VAUqJ3QCRL(1).jpg](https://github.com/user-attachments/assets/f5dab286-cdb7-40ed-98dc-ed385763dafa)

**Comment by DantesNoWar** [comment]
@nick-77 I've tried but this doesn't solve the problem. Some other ideas?

**Comment by GinnyGlider** [comment]
In the following screenshots I've switched the automatic repo sync from "Only on Wi-Fi" To "Always"
This has temporally solved a long-standing issue I had when using Droid-ify with a VPN ([I'm using](https://github.com/wgtunnel/wgtunnel)).

![Image](https://github.com/user-attachments/assets/4e208194-3467-4f28-8aca-3eb04aecbc9d)
![Image](https://github.com/user-attachments/assets/5853f24c-b0f8-4622-980d-4aaf254355c5)

My theory is Droid-ify mistakenly associating a VPN connection as a mobile connection. ¯\\_(ツ)_/¯

**Comment by DantesNoWar** [comment]
@GinnyGlider I'm running it with "always" already set up since months. So it's not the issue.
I'll test another client and see if the problem is related to certain repos or to Droidify only.

**Comment by Iamlooker** [comment]
@DantesNoWar did you find out if the issue is regarding Droid-ify or repo?

**Comment by DantesNoWar** [comment]
I don't know, didn't had time to check the source of the errors. Everytime I check there's few repos that doesn't sync, but htey're not always the same repos.
Also I've find out that Brave Beta and Nightly private F-droid repo fails to sync both on Droid-ify and NeoStore, so I assumed that the problem is on their side but I was not sure. So I deleted the repos in bot clients and reinstalled them, but the clients weren't able to connect to connect and fetch the repos.
